### PR TITLE
TEP: TON DNS Standard

### PR DESCRIPTION
**Pull request by v-lebedev**
TON DNS is a service for translating human-readable domain names (such as `test.ton` or `mysite.temp.ton`) into TON smart contract addresses, ADNL addresses employed by services running in the TON Network (such as TON Sites), and so on.

Moved from [TIP-81](https://github.com/ton-blockchain/TIPs/issues/81).

### CODE REVIEW

**Comment by Mamad3426m** [comment]
- [ ]

**Review by MrG9090 — APPROVED** [review]

File: text/0081-dns-standard.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+Bytes in range `0..32` (null character, control codes and space) inclusive are not allowed in domain names.
+
+For instance, `test.ton` and `mysite.temp.ton` are valid TON DNS domains.
+
+Technically, TON domains are case-sensitive, but TON apps and services convert all domains to lowercase before performing a TON DNS lookup in order to obtain case-insensitivity, so it makes no sense to register domains not in the lowercase.
+
+Note that a particular smart contract implementation may impose additional name limits when creating subdomains (for example, to avoid similar characters to protect against phishing). But the `dnsresolve` get-method must support the domain names in the format described above.
+
+## Domain internal representation
```
Comment (Emmydon049) [inline text/0081-dns-standard.md:38]:
```suggestion
## Domain internal representation 
```

Comment (Emmydon049) [inline text/0081-dns-standard.md:38]:
```suggestion
## Domain internal representation 
```

File: text/0081-dns-standard.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+The goal of this document is to collect feedback from community and to provide documentation for dApps developers.
+
+# Guide
+
+## Useful links
+1. [DNS Auction](https://dns.ton.org/) ([source code](https://github.com/ton-blockchain/dns))
+2. [ton.org documentation](https://ton.org/docs/#/web3/dns)
+3. Tolya answers about TON DNS (ru) - [1](https://github.com/ton-blockchain/TEPs/commit/4a09bfc737823f09f05dfb7008eec7784543bb2b), [2](https://telegra.ph/Otvety-na-voprosy-o-TON-DNS-kanalu-Investment-kingyru-CHast-2-08-06), [3](https://telegra.ph/Otvety-na-voprosy-o-TON-DNS-kanalu-Investment-kingyru-CHast-3-08-09)
+
+# Specification
```
Comment (Emmydon049) [inline text/0081-dns-standard.md:25]: Ton

Comment (Emmydon049) [inline text/0081-dns-standard.md:25]: Ton

Comment (Emmydon049) [inline text/0081-dns-standard.md:25]: Ton

File: text/0081-dns-standard.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
+- **TEP**: [81](https://github.com/ton-blockchain/TEPs/pull/5)
```
Comment (Emmydon049) [inline text/0081-dns-standard.md:1]: Ton

**Review by Emmydon049 — COMMENTED** [review]
UQC0frez8mjXrGyOriL48AzYgjeoYLHA09nkAWI59howKqpE

File: text/0081-dns-standard.md (hunk 1 above)
Comment (Emmydon049) [reply text/0081-dns-standard.md:38]: Notcoin

Comment (Emmydon049) [reply text/0081-dns-standard.md:38]: Notcoin

**Comment by cumcum8197** [comment]
2f2e0c5be1af8202ae75f16adb2c967dc479368e

**Comment by cumcum8197** [comment]
2f2e0c5be1af8202ae75f16adb2c967dc479368e

**Pull request merged**